File: src/web/faber-app/src/app/modules/resumes/resume-response.ts
import { Person } from './builder/person/person-response';
import { Education } from './builder/education/education-response';
import { Experience } from './builder/experience/experience-response';
import { Skill } from './builder/skill/skill-response';
import { Language } from './builder/language/language-response';
import { Link } from './builder/link/link-response';
import { Course } from './builder/course/course-response';

export interface Resume {
  id: string;
  createdAt: string;
  person: Person | null;
  title: string | null;
  summary: string;
  localization: string;
  hobbies: string;
  experience: Experience[];
  educations: Education[];
  courses: Course[];
  links: Link[];
  skills: Skill[];
  languages: Language[];
}

File: src/api/Modules/Resumes/Faber.Modules.Resumes.Application/Features/Resumes/Shared/ResumeResponse.cs
namespace Faber.Modules.Resumes.Application.Features.Resumes.Shared;

public record ResumeResponse(
    Guid Id,
    DateTimeOffset CreatedAt,
    string? Title,
    string? Localization,
    string? Summary,
    string? Hobbies,
    PersonResponse? Person,
    List<ExperienceResponse> Experience,
    List<EducationResponse> Educations,
    List<SkillResponse> Skills,
    List<LanguageResponse> Languages,
    List<CourseResponse> Courses,
    List<ProjectResponse> Projects,
    List<LinkResponse> Links);
